crates/transaction-pool/src/pool/mod.rs

```diff
@@ -896,7 +907,7 @@ where
     ///
     /// If some of the transactions are blob transactions, they are also removed from the blob
     /// store.
-    pub(crate) fn discard_worst(&self) -> HashSet<TxHash> {
```

Comment:
Unrelated change but this sets up for the micro-optimization mentioned above. And it's easier to optimize this whole thing if we can limit the `pub(crate)` scope.